tls/s2n_connection.c

```diff
@@ -576,7 +577,12 @@ int s2n_connection_wipe(struct s2n_connection *conn)
     /* Truncate the message buffers to save memory, we will dynamically resize it as needed */
     POSIX_GUARD(s2n_free(&conn->client_hello.raw_message));
     POSIX_GUARD(s2n_stuffer_resize(&conn->in, 0));
-    POSIX_GUARD(s2n_stuffer_resize(&conn->out, 0));
+
+    /* If the send buffer size has been set to non-zero value then the out stuffer
+     * was allocated to a specific size and was not growable. */
```

Comment:
Don't we still need to wipe the stuffer in the other case? There may be records in there that haven't been written to the socket, right?

Reply:
I can't resize the stuffer because we are tainting it with a raw write when we write records to it. Because the stuffer gets tainted I elected to make it statically sized since you cannot make a tainted stuffer grow.

This worked with the original logic because the stuffer get's wiped on every record.

This does make me think I should at least wipe and free the buffer. I'll add that